src/ploneintranet/search/example.py

```diff
@@ -41,16 +41,31 @@ def query(self, keywords, facets=None, start=0, step=None):
         facets.update({
             'SearchableText': keywords
         })
-        results = catalog.searchResults(facets)
-        batch = Batch(results, step, start)
+        brains = catalog.searchResults(facets)
+        batch = Batch(brains, step, start)
+        results = []
+        for brain in batch:
+            if brain['has_thumbs']:
+                preview_path = brain.getPath() + '/docconv_image_thumb.jpg'
+            else:
+                preview_path = None
+
+            results.append(
+                SearchResult(
+                    brain['Title'],
+                    brain.getPath(),
+                    description=brain['Description'],
+                    document_type=brain['friendly_type_name'],
+                    preview_path=preview_path,
+                )
+            )
         response_facets = {
-            'content_type': {x.portal_type for x in results},
-            'Subject': {y for x in results for y in x['Subject']},
+            'friendly_type_name': {x['friendly_type_name'] for x in brains},
```

Comment:
It appears that we're doing two iterations through the entire result set here. As this is only an example, it's probably fine but this is something we will certainly have to avoid in the real implementation.